lib/experimental/interactivity-api/directives/wp-class.php

```diff
@@ -0,0 +1,36 @@
+<?php
+/**
+ * Process wp-class directive attribute.
+ *
+ * @package Gutenberg
+ * @subpackage Interactivity API
+ */
+
+/**
+ * Process wp-class directive attribute.
+ *
+ * @param WP_Directive_Processor $tags Tags.
+ * @param WP_Directive_Context   $context Directive context.
+ */
+function gutenberg_interactivity_process_wp_class( $tags, $context ) {
+	if ( $tags->is_tag_closer() ) {
+		return;
+	}
+
+	$prefixed_attributes = $tags->get_attribute_names_with_prefix( 'data-wp-class--' );
+
+	foreach ( $prefixed_attributes as $attr ) {
+		list( , $class_name ) = explode( '--', $attr );
+		if ( empty( $class_name ) ) {
+			continue;
+		}
```

Comment:
how about a single function somewhere, maybe in `class-wp-directive-processor.php`, that parses the directive name. it's not much code, but given that we're already doing it in different ways and in each of these files, seems like it could be clearer to move it inward

```php
// class-wp-directive-processor {
	public static function parse_attribute_name( $name ) {
		/*
		 * Blah blah blargy bloop.
		 *
		 * Example
		 *     'wp-island'            => array( 'wp-island', null )
		 *     'wp-bind--src'         => array( 'wp-bind', 'src' )
		 *     'wp-thing--and--thang' => array( 'wp-thing', 'and--thang' )
		 */
		return explode( '--', $name, 2 );
	}
}
```

```php
list( $type ) = WP_Directive_Processor::parse_attribute_name( $name );

list( $type, $class_name ) = WP_Directive_Processor::parse_attribute_name( $name );
```